bf2fc6cc711aee1a0c2a/kas-fleet-manager
# abstract cluster management implementation

## Description


As part of https://issues.redhat.com/browse/MGDSTRM-1087, this PR adds a new doc to describe how we should abstract the implementation for OCM.

## Verification Steps


* Make sure the unit and integration tests are passing
* Checkout the branch and build it and run it:
    ```
     ./kas-fleet-manager serve --enable-kafka-external-certificate=true --mas-sso-enable-auth=true --public-host-url=<replace this with your own ngrok url> --dataplane-cluster-scaling-type=auto 
    ```
    You should see a cluster is created on OCM. Wait for the cluster to be provisioned successfully. The cluster should evetually reach the ready state.
* Stop the service, reset the database and modify the [configuration file](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/config/dataplane-cluster-configuration.yaml) to add a cluster. For example:
    ```
    clusters:
      - name: test
         cluster_id: <your cluster id, you can use the one from the previous step>
         cloud_provider: aws
         region: us-east-1
         schedulable: true
         kafka_instance_limit: 2
         status: cluster_provisioning
    ```
    and then start the kas-fleet-manager with the following command:
    ```
    ./kas-fleet-manager serve --enable-kafka-external-certificate=true --mas-sso-enable-auth=true --public-host-url=<replace this with your own ngrok url> --dataplane-cluster-scaling-type=manual 
    ```
    and you should see the cluster is provisioned successfully as well.
* Stop the service, remove the cluster from the configuration file and start the service again with the command from the previous step. You should see the cluster is deprovisoned, and cleaned up after the cluster is removed.

## Type of change

- [x] Refactoring (backward compatible)


## Checklist (Definition of Done)
